File: Java/217.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer,Integer> ref = new HashMap<>();
        
        for(int i:nums)
        {
            if(ref.containsKey(i))
                return true;
            ref.put(i,1);
        }
        return false;
    }
}

File: C++/217.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int, int> ref;
        for(int i = 0; i < nums.size(); i++)
        {
            if(ref.find(nums[i]) != ref.end())
                return true;
            ref[nums[i]] = 1;
        }
        return false;
    }
};